### PR TITLE
ci: format Julia code to Blue style, add CI to check new commits

### LINKED ISSUE #152: Refactor REISE.jl to comply with a standard style guide
# :rocket:

- [ ] Is your feature request essential for your project?

### Describe the workflow you want to enable
There would be no change to how users interact with `REISE.jl` or how `REISE.jl` functions.

### Describe your proposed implementation
Refactor the modules of `REISE.jl` to follow a single standard style guide. In the past, we have discussed following the [Blue](https://github.com/invenia/BlueStyle#documentation) style guide.

### PR DESCRIPTION
**Pull request by danielolsen**
[Pull Request doc](https://breakthrough-energy.github.io/docs/user/git_guide.html#d-pull-request)

### Purpose
Closes #152.

### What the code is doing
- Existing code is reformatted (should be no change to functionality).
- A `.git-blame-ignore-revs` file is added to ignore the reformat commit when using git blame.
- `lint.yml` is modified to add an automated check of Julia formatting.
- `.JuliaFormatter.toml` specifies that the automated check is against the Blue style (unlike black, which is the reformatter and the style, `JuliaFormatter` is a reformatter which can use one of many styles, so we have to tell it that we want `Blue`).

### Testing
Tested via pushes to this branch.

### Usage Example
```julia
using JuliaFormatter
format(".")
```

### Time estimate
15 minutes.

### CODE REVIEW

File: .github/workflows/lint.yml
Comment (jenhagg) [inline .github/workflows/lint.yml]: Looks like `julia-arch` isn't used anywhere. I'd say we can also just hard code the `runs-on: ubuntu-latest` (we probably won't be running formatting on multiple OSes).

Comment (danielolsen) [reply .github/workflows/lint.yml]: Done.

File: .github/workflows/lint.yml
Comment (jenhagg) [inline .github/workflows/lint.yml]: We use julia 1.5 in the [test.yml](https://github.com/Breakthrough-Energy/REISE.jl/blob/develop/.github/workflows/test.yml) workflow, should we stick with that for consistency?

Comment (danielolsen) [reply .github/workflows/lint.yml]: Done.

**Review by jenhagg — APPROVED** [review]
LGTM

**Comment by danielolsen** [comment]
I'll hold off on merging this until #154 and #155 are merged, since it will probably be easier for me to re-run the formatter than it would be to rebase all the new features onto the reformatted code.

**Comment by BainanXia** [comment]
This is nice. Two quick questions:
- What is the local setup so that we can run the same formatting tool locally before pushing code up into the repo?
    - I presume we can simply do the following:
    ```julia
  julia  -e 'using Pkg; Pkg.add(PackageSpec(name="JuliaFormatter"))'
  julia  -e 'using JuliaFormatter; format(".", verbose=true)'
    ``` 
- How frequently does `JuliaFormatter` get updated? Just wondering whether we need to keep an eye on updating the linting tool locally since the Git remote is always using the latest.

**Comment by danielolsen** [comment]
> This is nice. Two quick questions:
> 
>     * What is the local setup so that we can run the same formatting tool locally before pushing code up into the repo?
>       
>       * I presume we can simply do the following:
>       
>       ```julia
>       julia  -e 'using Pkg; Pkg.add(PackageSpec(name="JuliaFormatter"))'
>       julia  -e 'using JuliaFormatter; format(".", verbose=true)'
>       ```
> 
>     * How frequently does `JuliaFormatter` get updated? Just wondering whether we need to keep an eye on updating the linting tool locally since the Git remote is always using the latest.

1. Yes. You can run those from the command-line at the repo root, so that JuliaFormatter will work on all sub-directories and pick up the `.JuliaFormatter.toml` file to know to use blue, or you can run from any directory by using `using JuliaFormatter; format(PATH_OR_FILE, BlueStyle())`
2. JuliaFormatter seems to have a new minor release (major-ish, since it's `v0.x.y`) every two to three months. The Blue style guide doesn't have traditional releases, and changes to the README document seem to be pretty infrequent, so I think JuliaFormatter will be the one we want to keep an eye on. You could amend your command-line julia example above to include a new line between the first two: `julia  -e 'using Pkg; Pkg.update(PackageSpec(name="JuliaFormatter"))'`, which should make sure you always have the latest locally.

**Review by BainanXia — APPROVED** [review]
Nice.

**Comment by danielolsen** [comment]
I added some information to the README about the Blue style, and contributions more generally, per @dmuldrew's recommendation.

**Pull request merged**
